src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -0,0 +1,270 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file DataContext.hpp
+ */
+
+#ifndef GEOS_DATAREPOSITORY_DATACONTEXT_HPP_
+#define GEOS_DATAREPOSITORY_DATACONTEXT_HPP_
+
+#include "common/DataTypes.hpp"
+#include "common/Logger.hpp"
+#include "xmlWrapper.hpp"
+#include "common/Format.hpp"
+
+namespace geos
+{
+
+namespace dataRepository
+{
+
+class Group;
+class WrapperBase;
+
+/// DataContext is an abstract class storing contextual information on an object:
+/// - its line position in a file, if applicable, implementation in DataFileContext,
+/// - its location in the data hierarchy, implementation in GroupContext and WrapperContext.
+/// Typically, the target object contains an unique_ptr< DataContext > instance of this class.
+class DataContext
+{
+public:
+
+  /**
+   * @brief Construct a new DataContext object.
+   * @param objectName the target object name
+   * @param isDataFileContext true if this Context is a DataFileContext
+   */
+  DataContext( string const & objectName, bool isDataFileContext );
+
+  /**
+   * @brief Destroy the DataContext object
+   */
+  virtual ~DataContext() {}
+
+  /**
+   * @return A string that mention all the known informations to retrieve from where the target
+   * object comes from.
+   */
+  virtual string toString() const = 0;
+
+  /**
+   * @brief Get the target object name
+   */
+  string getObjectName() const
+  { return m_objectName; }
+
+  /**
+   * @brief Flag on availability of file information. Used to provide more user-friendly information.
+   * @return true if the context is from a file.
+   */
+  bool isDataFileContext() const
```

Comment:
I think we'll have to carefully have a look at this.

Reply:
What problem do you see here?